### DISCUSSION TITLE
Implementing an Anonymous Authentication

### DISCUSSION THREAD

**Discussion by adanielyan** [directus/directus#9566] [Feature Requests]
I want to implement an anonymous authentication in my app. It will allow users who haven't yet explicitly created an account in my web app to create content, add items to cart etc. Some OAuth providers already have this feature (https://firebase.google.com/docs/auth/web/anonymous-auth). 

Since Directus supports authentication by token, I was thinking I could silently create a Directus local user account for a user when one of a few predefined events happens (e.g. an item is added to cart or even when the app loads first time). This anonymous user account will only have a unique token that will be also stored in user's browser and will be added to headers as a Bearer token in all API requests to Directus.

The challenge is that once a user decides to explicitly create an account, his anonymous account must be converted into a regular one. I believe currently it can be achieved by implementing a local user creation logic in my app's backend, instead of calling Directus' API directly from the front end. Alternatively I can do it in a Directus custom API endpoint. So I can retrieve that user's anonymous account from Directus by token, modify it by adding a username, email, and password, removing the token, and saving it. Any further authentication will be done in a regular way. 

However, I would love to see this feature supported by Directus itself, instead if implementing a workaround solution in my app's backend. A dedicated API endpoint for anonymous auth would be a nice option. 

Another issue is handling user creation through a third party OAuth when they are enabled in Directus and AUTH_<PROVIDER>_ALLOW_PUBLIC_REGISTRATION is set to true. I think it may be again handled in my app's backend by implementing a user auth with a third party provider (e.g. GitHub) and then creating an account for a user as above. But then I'll have to handle 2 different OAuth flows: 1) done by my app, if user is anonymously authenticated but hasn't explicitly created an account yet, and 2) done by Directus if user already has an account and should be only authenticated by a third party OAuth provider. This is possible, but will add a lot of additional logic and code to my app. I think it would be great if directus had this feature built-in.

I will be happy to discuss this further and share my thoughts if this is something that others are interested in too.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by kellvembarbosa** [reply]
Did that die? Don't you even have plans for the future? It's something so important for mobile applications...

**Reply by br41nslug** [reply]
Unfortunately this request has only gotten only 6 upvotes in almost 2 years, indicating to us that the community does not have enough interest in this feature to be implemented at this time.

Feel free to recreate the feature request and if this feature is important enough it should be very doable to get to those 15 upvotes.
